modules/hal_rpi_pico/bootloader/CMakeLists.txt

```diff
@@ -0,0 +1,51 @@
+# SPDX-License-Identifier: Apache-2.0
+
+cmake_minimum_required(VERSION 3.20.0)
+
+# Skip compiler checking
+set(CMAKE_TRY_COMPILE_TARGET_TYPE STATIC_LIBRARY)
+
+project(second_stage_bootloader)
+enable_language(ASM)
+
+set(rp2_common_dir ${ZEPHYR_HAL_RPI_PICO_MODULE_DIR}/src/rp2_common)
+set(rp2040_dir ${ZEPHYR_HAL_RPI_PICO_MODULE_DIR}/src/rp2040)
+set(common_dir ${ZEPHYR_HAL_RPI_PICO_MODULE_DIR}/src/common)
+set(boot_stage_dir ${rp2_common_dir}/boot_stage2)
+
+add_executable(boot_stage2)
+
+target_sources(boot_stage2 PRIVATE ${boot_stage_dir}/${FLASH_TYPE_FILE})
```

Comment:
Here, use the `FLASH_TYPE` information that is passed, and then select the correct source file.